Bottom: 90c1b0ee81564347722286b277a390893378ef08
Top:    01b3621b5a7c96c318330579e015bdf5499a8a52
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-19 01:00:23 +0000

src/action.c: Abort the Import(setnewport,mask) action if the mask isn't set

This avoids using an uninitialised variable.

(Caught by clang)


---

diff --git a/src/action.c b/src/action.c
index c1325ba..1005b17 100644
--- a/src/action.c
+++ b/src/action.c
@@ -7622,11 +7622,11 @@ ActionImport (int argc, char **argv, Coord x, Coord y)
 	}
       else if (strcasecmp (xs, "mark") == 0)
 	{
-	  if (Marked.status)
-	    {
-	      x = Marked.X;
-	      y = Marked.Y;
-	    }
+	  if (!Marked.status)
+	    return 0;
+
+	  x = Marked.X;
+	  y = Marked.Y;
 	}
       else if (ys)
 	{
